### PR TITLE
[#315] Reset password handlig fix

### PR DESCRIPTION
**Pull request by kmarszalek**
- add deleting preexisting token while requesting a new one
- add token expiry verification
- add tests
- removed unnecessary content type in tests on get endpoints

### CODE REVIEW

File: s4e-backend/src/test/java/pl/cyfronet/s4e/service/PasswordServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+                .expiryTimestamp(LocalDateTime.now().plus(EXPIRE_IN))
+                .build();
+        passwordResetRepository.save(token);
+        try {
+            passwordService.validate(token.getToken());
+        } catch (NotFoundException e) {
+            fail("Token not found");
+        } catch (PasswordResetTokenExpiredException e) {
+            fail("Token expired");
+        }
```
Comment (jswk) [inline s4e-backend/src/test/java/pl/cyfronet/s4e/service/PasswordServiceTest.java]: Remove the try-catch here. If it throws, the test will fail anyway. Alternatively, pass the exception as the second argument of the `fail` call.

File: s4e-backend/src/test/java/pl/cyfronet/s4e/service/PasswordServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+        assertThat(passwordService.findByEmail(PROFILE_EMAIL), isEmpty());
+        passwordService.createPasswordResetTokenForUser( PROFILE_EMAIL);
+        assertThat(passwordService.findByEmail(PROFILE_EMAIL), isPresent());
+    }
+
+    @Test
+    public void shouldntCreatePasswordResetTokenForUser() {
+        assertThat(passwordService.findByEmail(PROFILE_EMAIL), isEmpty());
+        assertThrows(NotFoundException.class, () -> passwordService.createPasswordResetTokenForUser( "mail@email.pl"));
+        assertThat(passwordService.findByEmail(PROFILE_EMAIL), isEmpty());
```
Comment (jswk) [inline s4e-backend/src/test/java/pl/cyfronet/s4e/service/PasswordServiceTest.java]: You should assert that `findByEmail("mail@email.pl")` return empty, not `findByEmail(PROFILE_EMAIL)`.

File: s4e-backend/src/test/java/pl/cyfronet/s4e/service/PasswordServiceTest.java (hunk 1 above)
Comment (kmarszalek) [reply s4e-backend/src/test/java/pl/cyfronet/s4e/service/PasswordServiceTest.java]: yeah, not sure how that happen :D

File: s4e-backend/src/test/java/pl/cyfronet/s4e/service/PasswordServiceTest.java (hunk 2 above)
Comment (kmarszalek) [reply s4e-backend/src/test/java/pl/cyfronet/s4e/service/PasswordServiceTest.java]: nice catch :)

**Pull request merged**